elixir/test/contracts_test.exs

```diff
@@ -3,90 +3,417 @@ defmodule Trento.ContractsTest do
 
   alias CloudEvents.CloudEvent
 
-  test "should decode to the right struct" do
-    event_id = UUID.uuid4()
-    event = %Test.Event{id: event_id}
-
-    cloudevent = %CloudEvent{
-      data:
-        {:proto_data,
-         %Google.Protobuf.Any{
-           __unknown_fields__: [],
-           type_url: "test.Event",
-           value: Test.Event.encode(event)
-         }},
-      id: UUID.uuid4(),
-      source: "wandalorian",
-      spec_version: "1.0",
-      type: "test.Event"
-    }
-
-    encoded_cloudevent = CloudEvent.encode(cloudevent)
-
-    assert {:ok, %Test.Event{id: ^event_id}} = Trento.Contracts.from_event(encoded_cloudevent)
-  end
+  describe "message with signature" do
```

Comment:
nitpick: maybe `signed message`. and the other describe could be `unsigned message`